src/mono/wasi/runtime/driver.c

```diff
@@ -11,6 +11,7 @@
 
 #include <mono/metadata/appdomain.h>
 #include <mono/metadata/assembly.h>
+#include <mono/metadata/bundled-resources-internals.h>
```

Comment:
Looks like the driver links towards runtime libraries, so in that case I assume it is primarily using public headers and should not include bundled-resources-internals.h.

Instead do extern declare of used mono_bundled_resources_get_data_resource_values, similar pattern as done previously.